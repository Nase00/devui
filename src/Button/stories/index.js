import React from 'react';
import { storiesOf, action } from '@kadira/storybook';
import styled from 'styled-components';
import { withKnobs, text, boolean, select } from '@kadira/storybook-addon-knobs';
import MdFiberManualRecord from 'react-icons/lib/md/fiber-manual-record';
import Button from '../';

export const Container = styled.div`
  display: flex;
  height: 100%;
  width: 100%;
  justify-content: center;
  align-items: center;

  .wrapper {
    display: flex;
    width: 100%;
    padding: 5px;
    background-color: ${props => props.theme.base01};
    text-align: center;
  }
`;

storiesOf('Button', module)
  .addDecorator(withKnobs)
  .addWithInfo(
    'default',
    '',
    () => (
      <Container>
        <Button
          title={text('Title', 'Hello Tooltip')}
          tooltipPosition={
            select('tooltipPosition', ['top', 'bottom', 'left', 'right',
             'bottom-left', 'bottom-right', 'top-left', 'top-right'])
          }
          big={boolean('Big', true)}
          disabled={boolean('Disabled', false)}
          onClick={action('button clicked')}
        >
          {text('Label', 'Hello Button')}
        </Button>
      </Container>
    )
  )
  .addWithInfo(
    'mark',
    '',
    () => (
      <Container>
        <Button
          mark={select('mark', [false, 'base08', 'base09', 'base0A', 'base0B',
            'base0C', 'base0D', 'base0E', 'base0F'], 'base08')}
          title={text('Title', 'Hello Tooltip')}
          tooltipPosition={
            select('tooltipPosition', ['top', 'bottom', 'left', 'right',
             'bottom-left', 'bottom-right', 'top-left', 'top-right'])
          }
          big={boolean('Big', true)}
          disabled={boolean('Disabled', false)}
          onClick={action('button clicked')}
        >
          <MdFiberManualRecord />
        </Button>
      </Container>
    )
  );
