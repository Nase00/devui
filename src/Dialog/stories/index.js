import React from 'react';
import { storiesOf, action } from '@kadira/storybook';
import { withKnobs, text, boolean } from '@kadira/storybook-addon-knobs';
import Dialog from '../';

storiesOf('Dialog', module)
  .addDecorator(withKnobs)
  .addWithInfo(
    'default',
    '',
    () => (
      <Dialog
        title={text('title', 'Dialog Title')}
        children={text('children', 'Hello Dialog!')}
        submitText={text('submitText', 'Submit!')}
        open={boolean('open', true)}
        modal={boolean('modal', false)}
        fullWidth={boolean('fullWidth', false)}
        onDismiss={action('dialog dismissed')}
        onSubmit={action('dialog submitted')}
      />
    )
  );
