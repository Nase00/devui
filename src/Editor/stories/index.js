import React from 'react';
import { storiesOf } from '@kadira/storybook';
import { withKnobs, text, select, boolean } from '@kadira/storybook-addon-knobs';
import Editor from '../';
import WithTabs from './WithTabs';

const themes = {
  default: 'default',
  night: 'night',
  'solarized dark': 'solarized dark',
  'solarized light': 'solarized light',
  'tomorrow-night-bright': 'tomorrow-night-bright'
};

const value = `
var themes = ${JSON.stringify(themes, null, 2)};

function getThemes() {
  return themes;
}
`;

storiesOf('Editor', module)
  .addDecorator(withKnobs)
  .addWithInfo(
    'default',
    '',
    () => (
      <Editor
        theme={select('theme', themes, 'default')}
        value={text('value', value)}
        lineNumbers={boolean('lineNumbers', true)}
        lineWrapping={boolean('lineWrapping', false)}
        foldGutter={boolean('foldGutter', true)}
        readOnly={boolean('readOnly', false)}
        autofocus
      />
    )
  )
  .addWithInfo(
    'with tabs',
    '',
    () => (
      <WithTabs
        value={text('value', value)}
        lineNumbers={boolean('lineNumbers', true)}
      />
    )
  );
