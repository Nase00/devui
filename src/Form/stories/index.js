import React from 'react';
import { storiesOf, action } from '@kadira/storybook';
import { withKnobs, object } from '@kadira/storybook-addon-knobs';
import Form from '../';
import { schema, uiSchema, formData } from './schema';

storiesOf('Form', module)
  .addDecorator(withKnobs)
  .addWithInfo(
    'default',
    'Wrapper around [`react-jsonschema-form`](https://github.com/mozilla-services/react-jsonschema-form) with custom widgets.',
    () => (
      <Form
        formData={object('formData', formData)}
        schema={object('schema', schema)}
        uiSchema={object('uiSchema', uiSchema)}
        onChange={action('form changed')}
        onSubmit={action('form submitted')}
      />
    )
  );
