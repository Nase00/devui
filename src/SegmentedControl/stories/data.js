import React from 'react';

/* eslint-disable react/prop-types */
const Component = ({ selected }) => (
  <div
    style={{
      display: 'flex',
      alignItems: 'center',
      justifyContent: 'center',
      width: '100%',
      height: '100%',
      fontSize: '22px'
    }}
  >
    Selected {selected}
  </div>
);
/* eslint-enable react/prop-types */

const selector = button => ({ selected: button.name });

export const buttons = [
  {
    name: 'Button1',
    component: Component,
    selector
  },
  {
    name: 'Button2',
    component: Component,
    selector
  },
  {
    name: 'Button3',
    component: Component,
    selector
  }
];

export const simple10Buttons = [];
for (let i = 1; i <= 10; i++) simple10Buttons.push({ name: `Button${i}`, value: `${i}` });
