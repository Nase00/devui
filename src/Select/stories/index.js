import React from 'react';
import { storiesOf, action } from '@kadira/storybook';
import { withKnobs, text, number, boolean } from '@kadira/storybook-addon-knobs';
import Select from '../';
import { options } from './options';

storiesOf('Select', module)
  .addDecorator(withKnobs)
  .addWithInfo(
    'default',
    'Wrapper around [React Select](https://github.com/JedWatson/react-select) with themes and new props like `openOuterUp` and `menuMaxHeight`.',
    () => (
      <div
        style={{
          display: 'flex',
          height: '100%', width: '95%', margin: 'auto',
          justifyContent: 'center', alignItems: 'center'
        }}
      >
        <Select
          value={text('value', 'one')}
          menuMaxHeight={number('menuMaxHeight', 200)}
          options={options}
          onChange={action('selected')}
          autosize={boolean('autosize', false)}
          clearable={boolean('clearable', false)}
          disabled={boolean('disabled', false)}
          isLoading={boolean('isLoading', false)}
          multi={boolean('multiselect', false)}
          searchable={boolean('searchable', true)}
          openOuterUp={boolean('openOuterUp', false)}
        />
      </div>
    )
  );
