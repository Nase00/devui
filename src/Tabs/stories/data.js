import React from 'react';

/* eslint-disable react/prop-types */
const Component = ({ selected }) => (
  <div
    style={{
      display: 'flex',
      alignItems: 'center',
      justifyContent: 'center',
      width: '100%',
      height: '100%',
      fontSize: '22px'
    }}
  >
    Selected {selected}
  </div>
);
/* eslint-enable react/prop-types */

const selector = tab => ({ selected: tab.name });

export const tabs = [
  {
    name: 'Tab1',
    component: Component,
    selector
  },
  {
    name: 'Tab2',
    component: Component,
    selector
  },
  {
    name: 'Tab3',
    component: Component,
    selector
  }
];
