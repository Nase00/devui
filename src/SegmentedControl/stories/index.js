import React from 'react';
import { storiesOf, action } from '@kadira/storybook';
import styled from 'styled-components';
import { withKnobs, text, boolean, select } from '@kadira/storybook-addon-knobs';
import SegmentedControl from '../';
import { tabs } from './data';

export const Container = styled.div`
  display: flex;
  height: 100%;
  width: 100%;
  justify-content: center;
  align-items: center;
`;

storiesOf('SegmentedControl', module)
  .addDecorator(withKnobs)
  .addWithInfo(
    'default',
    '',
    () => (
      <Container>
        <SegmentedControl
          disabled={boolean('Disabled', false)}
          onClick={action('tab selected')}
          tabs={tabs}
          align={select('align', ['left', 'right'])}
          selected={text('selected', 'Tab1')}
        />
      </Container>
    )
  );
