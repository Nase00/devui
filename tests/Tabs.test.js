import React from 'react';
import { render, mount } from 'enzyme';
import { renderToJson } from 'enzyme-to-json';
import Tabs from '../src/Tabs';
import { tabs } from '../src/Tabs/stories/data';

describe('Tabs', function () {
  it('renders correctly', () => {
    const wrapper = render(<Tabs tabs={tabs} onClick={() => {}} />);
    expect(renderToJson(wrapper)).toMatchSnapshot();
  });

  it('renders with props', () => {
    const wrapper = render(
      <Tabs
        tabs={tabs}
        onClick={() => {}}
        buttons={[
          <button key="button1">Button 1</button>,
          <button key="button2">Button 2</button>
        ]}
        compact selected="Tab2"
      />
    );
    expect(renderToJson(wrapper)).toMatchSnapshot();
  });

  it('should select tab', () => {
    const onClick = jest.fn();
    const wrapper = mount(<Tabs tabs={tabs} onClick={onClick} />);

    wrapper.find('button').first().simulate('click');
    expect(onClick).toBeCalled();
  });
});
