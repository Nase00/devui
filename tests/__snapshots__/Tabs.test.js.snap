exports[`Tabs renders correctly 1`] = `
<div
  class="foLpMM">
  <div
    class="jLtDvn">
    <div>
      <button
        data-selected="true"
        value="Tab1">
        Tab1
      </button>
      <button
        value="Tab2">
        Tab2
      </button>
      <button
        value="Tab3">
        Tab3
      </button>
    </div>
    <div />
  </div>
  <div>
    <div
      style="display:flex;align-items:center;justify-content:center;width:100%;height:100%;font-size:22px;">
      Selected Tab1
    </div>
  </div>
</div>
`;

exports[`Tabs renders with props 1`] = `
<div
  class="foLpMM">
  <div
    class="jLtDvn">
    <div>
      <button
        value="Tab1">
        Tab1
      </button>
      <button
        data-selected="true"
        value="Tab2">
        Tab2
      </button>
      <button
        value="Tab3">
        Tab3
      </button>
    </div>
    <div>
      <button>
        Button 1
      </button>
      <button>
        Button 2
      </button>
    </div>
  </div>
  <div>
    <div
      style="display:flex;align-items:center;justify-content:center;width:100%;height:100%;font-size:22px;">
      Selected Tab2
    </div>
  </div>
</div>
`;
